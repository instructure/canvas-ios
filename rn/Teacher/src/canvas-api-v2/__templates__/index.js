// @flow
export * from './submission'
export * from './user'
export * from './assignment'
export * from './course'
export * from './group'
export * from './section'
export * from './groupSet'

// QUERIES
export * from '../queries/__templates__/SubmissionList'
export * from '../queries/__templates__/ContextCard'
