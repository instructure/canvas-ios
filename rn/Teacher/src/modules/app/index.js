//
// Copyright (C) 2017-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

export type AppId = 'student' | 'teacher' | 'parent'
export type App = {
  appId: AppId,
  filterCourse: (course: Course) => boolean,
}

const teacher = {
  appId: 'teacher',
  filterCourse: (course: Course): boolean => {
    if (course.access_restricted_by_date) return false
    const enrollments = course.enrollments
    if (!enrollments) return false
    return enrollments.some((e) =>
      [
        'teacher',
        'teacherenrollment',
        'designer',
        'ta',
      ].includes(e.type.toLowerCase())
    )
  },
}

const student = {
  appId: 'student',
  filterCourse: (course: Course): boolean => !course.access_restricted_by_date,
}

const parent = {
  appId: 'parent',
  filterCourse: (course: Course): boolean => true,
}

let current: App = teacher

const app = {
  setCurrentApp: (appId: AppId): void => {
    switch (appId) {
      case 'student':
        current = student
        break
      case 'teacher':
        current = teacher
        break
      case 'parent':
        current = parent
        break
    }
  },
  current: (): App => current,
}

export function isTeacher (): boolean {
  return app.current().appId === 'teacher'
}
export function isStudent (): boolean {
  return app.current().appId === 'student'
}

export function isParent (): boolean {
  return app.current().appId === 'parent'
}

export default app
