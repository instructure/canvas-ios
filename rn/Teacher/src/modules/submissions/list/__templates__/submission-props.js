//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

import template, { type Template } from '../../../../utils/template'
import type { SubmissionDataProps } from '../submission-prop-types'

const templates = {
  ...require('../../../../__templates__/submissions'),
}

const submissionWithHistory = templates.submissionHistory()

// $FlowFixMe
export const submissionProps: Template<SubmissionDataProps> = template({
  grade: 'B-',
  name: submissionWithHistory.user.name,
  status: 'submitted',
  userID: submissionWithHistory.user.id,
  avatarURL: submissionWithHistory.user.avatar_url,
  sectionID: '1',
})
