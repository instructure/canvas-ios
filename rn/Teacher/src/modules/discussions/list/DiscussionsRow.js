//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

/* @flow */

import React, { PureComponent } from 'react'
import {
  View,
  TouchableHighlight,
  Image,
} from 'react-native'
import i18n from 'format-message'
import { createStyleSheet } from '../../../common/stylesheet'

import Row from '../../../common/components/rows/Row'
import AccessIcon from '../../../common/components/AccessIcon'
import { formattedDueDateWithStatus, formattedDueDate } from '../../../common/formatters'
import { extractDateFromString } from '../../../utils/dateUtils'
import instIcon from '../../../images/inst-icons'
import { DotSeparated, Text } from '../../../common/text'
import { isTeacher } from '../../app'

export type Props = {
  discussion: Discussion,
  onPress: (discussion: Discussion) => void,
  index: number,
  tintColor: ?string,
  onToggleDiscussionGrouping: Function,
  selected: boolean,
}

export default class DiscussionsRow extends PureComponent<Props> {
  render () {
    const { selected, discussion } = this.props
    const points = this.points(discussion)
    const discussionDetails = this.discussionDetails(discussion)
    const unreadDot = this.renderUnreadDot(discussion)
    return (
      <View>
        <View accessible={false}>
          <Row accessible={true}
            renderImage={this.renderIcon}
            title={discussion.title}
            titleProps={{ ellipsizeMode: 'tail', numberOfLines: 2 }}
            border='bottom'
            disclosureIndicator={false}
            testID={`DiscussionListCell.${discussion.id}`}
            onPress={this.onPress}
            height='auto'
            selected={selected}
          >
            <View style={style.rowContent}>
              <View style={style.mainContentColumn} accessible={false}>
                <DotSeparated style={style.subtitle} separated={this.dueDate(discussion)}/>

                {points &&
                  <View testID='discussion.row.points' style={style.details}>
                    <Text style={style.points}>{points}</Text>
                  </View>
                }

                {discussionDetails &&
                  <View testID='discussion.row.details' style={style.details}>
                    {discussionDetails}
                  </View>
                }
              </View>
              { isTeacher() && this.renderKabob() }
            </View>
          </Row>

        </View>
        { unreadDot }
      </View>
    )
  }

  renderKabob = () => {
    const discussion = this.props.discussion
    return (
      <TouchableHighlight
        style={style.kabobButton}
        onPress={this.onToggleDiscussionGrouping}
        accessibilityTraits='button'
        accessible={true}
        accessibilityLabel={i18n('Change {discussionTitle} to different grouping', { discussionTitle: discussion.title })}
        underlayColor='#ffffff00'
        testID={`discussion.kabob-${this.props.discussion.id}`}
      >
        <Image style={style.kabob} source={instIcon('more')}/>
      </TouchableHighlight>
    )
  }

  onPress = () => {
    this.props.onPress(this.props.discussion)
  }

  onToggleDiscussionGrouping = () => {
    this.props.onToggleDiscussionGrouping(this.props.discussion)
  }

  dueDate = (discussion: Discussion) => {
    const { due_at, lock_at, has_overrides: multiple } = (discussion.assignment || {})
    const { last_reply_at } = discussion
    const dueAt = extractDateFromString(due_at)
    const lockAt = extractDateFromString(lock_at)
    const lastReply = extractDateFromString(last_reply_at)
    if (!dueAt && lastReply) {
      let lastReplyDateStr = formattedDueDate(lastReply)
      let lastPostAt = i18n('Last post {lastReplyDateStr}', { lastReplyDateStr })
      return [lastPostAt]
    }
    if (isTeacher() && dueAt && multiple) {
      return [i18n('Multiple Due Dates')]
    }
    return formattedDueDateWithStatus(dueAt, lockAt)
  }

  renderUnreadDot = (discussion: Discussion) => {
    if (discussion.unread_count > 0) {
      return (<View testID='discussions.row.unread-dot' style={style.unreadDot}/>)
    } else {
      return (<View testID='discussions.row.unread-dot' />)
    }
  }

  renderIcon = () => {
    const { discussion } = this.props
    return (
      <View style={style.icon}>
        <AccessIcon
          entry={discussion}
          tintColor={this.props.tintColor}
          image={instIcon(discussion.assignment ? 'assignment' : 'discussion')}
        />
      </View>
    )
  }

  discussionDetails (discussion: Discussion) {
    const replies = i18n({
      default: `{
        count, plural,
        one {# Reply}
        other {# Replies}
      }`,
      description: 'Number of replies',
    }, { count: discussion.discussion_subentry_count })
    const unread = i18n({
      default: `{
        unread, plural,
          one {# Unread}
        other {# Unread}
      }`,
      description: 'Number of unread discussion posts',
    }, { unread: discussion.unread_count })

    return <DotSeparated style={style.discussionDetails} separated={[replies, unread].filter(v => v)}/>
  }

  points = (discussion: Discussion) => {
    if (discussion.assignment) {
      const pointsPossible = Boolean(discussion.assignment.points_possible) && i18n({
        default: `{
        count, plural,
        one {# pt}
        other {# pts}
      }`,
        description: 'Number of points possible',
      }, { count: discussion.assignment.points_possible })
      return pointsPossible
    }
  }
}

const unreadDotSize = 5
const style = createStyleSheet((colors, vars) => ({
  rowContent: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
  },
  mainContentColumn: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
  },
  details: {
    flexDirection: 'row',
    paddingTop: 2,
  },
  icon: {
    alignSelf: 'flex-start',
  },
  subtitle: {
    color: colors.textDark,
    fontSize: 14,
    marginTop: 2,
  },
  points: {
    fontSize: 14,
  },
  discussionDetails: {
    fontSize: 14,
  },
  unreadDot: {
    width: unreadDotSize,
    height: unreadDotSize,
    borderRadius: unreadDotSize / 2,
    backgroundColor: colors.textInfo,
    position: 'absolute',
    top: 6,
    left: 8,
  },
  kabobButton: {
    flex: 0,
    flexDirection: 'column',
    justifyContent: 'center',
    alignSelf: 'center',
    width: 43,
    height: 43,
    marginTop: -15,
    marginRight: -vars.padding / 2,
  },
  kabob: {
    alignSelf: 'center',
    width: 18,
    height: 18,
    tintColor: colors.textDark,
    transform: [{ rotate: '180deg' }],
  },
}))
