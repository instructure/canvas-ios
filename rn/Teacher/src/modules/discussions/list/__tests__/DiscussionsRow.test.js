//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow
import { shallow } from 'enzyme'
import React from 'react'
import * as template from '../../../../__templates__'
import app from '../../../app'
import DiscussionsRow from '../DiscussionsRow'

describe('DiscussionsRow', () => {
  let props
  beforeEach(() => {
    app.setCurrentApp('teacher')
    props = {
      discussion: template.discussion(),
      onPress: jest.fn(),
      index: 0,
      tintColor: '#fff',
      onToggleDiscussionGrouping: jest.fn(),
      selected: false,
    }
  })

  it('renders', () => {
    const tree = shallow(<DiscussionsRow {...props} />)

    expect(tree.find('Row').prop('title')).toEqual(props.discussion.title)
    expect(tree.find('Row').prop('selected')).toEqual(props.selected)
    expect(tree.find('DotSeparated').at(0).prop('separated')).toEqual(['Due May 31, 2037 at 11:59 PM'])
  })

  it('sends onPress', () => {
    const tree = shallow(<DiscussionsRow {...props} />)
    tree.find(`[testID="DiscussionListCell.${props.discussion.id}"]`).simulate('press')
    expect(props.onPress).toHaveBeenCalledWith(props.discussion)
  })

  it('renders without points possible', () => {
    props.discussion.assignment.points_possible = null
    const tree = shallow(<DiscussionsRow {...props} />)
    expect(tree.find('[testID="discussion.row.points"]').exists()).toEqual(false)
  })

  it('renders with points possible', () => {
    props.discussion.assignment.points_possible = 12
    const tree = shallow(<DiscussionsRow {...props} />)
    expect(tree.find('[testID="discussion.row.points"] Text').prop('children')).toEqual('12 pts')
  })

  it('shows multiple due dates for teacher when assignment has overrides', () => {
    app.setCurrentApp('teacher')
    props.discussion.assignment.has_overrides = true
    const tree = shallow(<DiscussionsRow {...props} />)
    expect(tree.find('DotSeparated').first().prop('separated')).toEqual([
      'Multiple Due Dates',
    ])
  })

  it('renders with no unread count', () => {
    props.discussion.unread_count = 0
    const tree = shallow(<DiscussionsRow {...props} />)
    expect(tree.find('[testID="discussion.row.details"] DotSeparated').prop('separated')).toEqual(['2 Replies', '0 Unread'])
  })

  it('renders with an unread count', () => {
    props.discussion.unread_count = 2
    const tree = shallow(<DiscussionsRow {...props} />)
    expect(tree.find('[testID="discussion.row.details"] DotSeparated').prop('separated')).toEqual(['2 Replies', '2 Unread'])
  })

  it('renders no replies', () => {
    props.discussion.discussion_subentry_count = 0
    const tree = shallow(<DiscussionsRow {...props} />)
    expect(tree.find('[testID="discussion.row.details"] DotSeparated').prop('separated')).toEqual(['0 Replies', '1 Unread'])
  })

  it('renders the unread dot', () => {
    props.discussion.unread_count = 2
    const tree = shallow(<DiscussionsRow {...props} />)
    expect(tree.find('[testID="discussions.row.unread-dot"]').prop('style')).not.toBeUndefined()
  })

  it('does not render the unread dot', () => {
    props.discussion.unread_count = 0
    const tree = shallow(<DiscussionsRow {...props} />)
    expect(tree.find('[testID="discussions.row.unread-dot"]').prop('style')).toBeUndefined()
  })

  it('renders with no assignment', () => {
    props.discussion.assignment = null
    const tree = shallow(<DiscussionsRow {...props} />)
    expect(tree.find('DotSeparated').at(0).prop('separated')).toEqual(['Last post Dec 10, 2016 at 9:03 PM'])
  })
})
