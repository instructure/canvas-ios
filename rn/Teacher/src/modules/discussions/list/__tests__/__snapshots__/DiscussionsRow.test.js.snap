// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscussionsRow renders 1`] = `
<View>
  <View
    accessible={false}
  >
    <Row
      accessible={true}
      border="bottom"
      disclosureIndicator={false}
      height="auto"
      onPress={[Function]}
      renderImage={[Function]}
      selected={false}
      testID="DiscussionListCell.1"
      title="Dear Ivy Iversen, your last name cannot start with your first name"
      titleProps={
        Object {
          "ellipsizeMode": "tail",
          "numberOfLines": 2,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={false}
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <DotSeparated
            separated={
              Array [
                "Due May 31, 2037 at 11:59 PM",
              ]
            }
            style={
              Object {
                "color": "#8B969E",
                "fontSize": 14,
                "marginTop": 2,
              }
            }
          />
          <View
            style={
              Object {
                "flexDirection": "row",
                "paddingTop": 2,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 14,
                }
              }
            >
              20 pts
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "paddingTop": 2,
              }
            }
          >
            <DotSeparated
              separated={
                Array [
                  "2 Replies",
                  "1 Unread",
                ]
              }
              style={
                Object {
                  "fontSize": 14,
                }
              }
            />
          </View>
        </View>
        <TouchableHighlight
          accessibilityLabel="Change Dear Ivy Iversen, your last name cannot start with your first name to different grouping"
          accessibilityTraits="button"
          accessible={true}
          activeOpacity={0.85}
          delayPressOut={100}
          onPress={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "flex": 0,
              "flexDirection": "column",
              "height": 43,
              "justifyContent": "center",
              "marginRight": -8,
              "marginTop": -15,
              "width": 43,
            }
          }
          testID="discussion.kabob-1"
          underlayColor="#ffffff00"
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/images/kabob.png",
              }
            }
            style={
              Object {
                "alignSelf": "center",
                "height": 18,
                "tintColor": "#8B969E",
                "transform": Array [
                  Object {
                    "rotate": "180deg",
                  },
                ],
                "width": 18,
              }
            }
          />
        </TouchableHighlight>
      </View>
    </Row>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#008EE2",
        "borderRadius": 2.5,
        "height": 5,
        "left": 8,
        "position": "absolute",
        "top": 6,
        "width": 5,
      }
    }
  />
</View>
`;

exports[`DiscussionsRow renders published 1`] = `
<View>
  <View
    accessible={false}
  >
    <Row
      accessible={true}
      border="bottom"
      disclosureIndicator={false}
      height="auto"
      onPress={[Function]}
      renderImage={[Function]}
      selected={false}
      testID="DiscussionListCell.1"
      title="Dear Ivy Iversen, your last name cannot start with your first name"
      titleProps={
        Object {
          "ellipsizeMode": "tail",
          "numberOfLines": 2,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={false}
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <DotSeparated
            separated={
              Array [
                "Due May 31, 2037 at 11:59 PM",
              ]
            }
            style={
              Object {
                "color": "#8B969E",
                "fontSize": 14,
                "marginTop": 2,
              }
            }
          />
          <View
            style={
              Object {
                "flexDirection": "row",
                "paddingTop": 2,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 14,
                }
              }
            >
              20 pts
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "paddingTop": 2,
              }
            }
          >
            <DotSeparated
              separated={
                Array [
                  "2 Replies",
                  "1 Unread",
                ]
              }
              style={
                Object {
                  "fontSize": 14,
                }
              }
            />
          </View>
        </View>
        <TouchableHighlight
          accessibilityLabel="Change Dear Ivy Iversen, your last name cannot start with your first name to different grouping"
          accessibilityTraits="button"
          accessible={true}
          activeOpacity={0.85}
          delayPressOut={100}
          onPress={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "flex": 0,
              "flexDirection": "column",
              "height": 43,
              "justifyContent": "center",
              "marginRight": -8,
              "marginTop": -15,
              "width": 43,
            }
          }
          testID="discussion.kabob-1"
          underlayColor="#ffffff00"
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/images/kabob.png",
              }
            }
            style={
              Object {
                "alignSelf": "center",
                "height": 18,
                "tintColor": "#8B969E",
                "transform": Array [
                  Object {
                    "rotate": "180deg",
                  },
                ],
                "width": 18,
              }
            }
          />
        </TouchableHighlight>
      </View>
    </Row>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#008EE2",
        "borderRadius": 2.5,
        "height": 5,
        "left": 8,
        "position": "absolute",
        "top": 6,
        "width": 5,
      }
    }
  />
</View>
`;

exports[`DiscussionsRow renders unpublished 1`] = `
<View>
  <View
    accessible={false}
  >
    <Row
      accessible={true}
      border="bottom"
      disclosureIndicator={false}
      height="auto"
      onPress={[Function]}
      renderImage={[Function]}
      selected={false}
      testID="DiscussionListCell.1"
      title="Dear Ivy Iversen, your last name cannot start with your first name"
      titleProps={
        Object {
          "ellipsizeMode": "tail",
          "numberOfLines": 2,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={false}
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <DotSeparated
            separated={
              Array [
                "Due May 31, 2037 at 11:59 PM",
              ]
            }
            style={
              Object {
                "color": "#8B969E",
                "fontSize": 14,
                "marginTop": 2,
              }
            }
          />
          <View
            style={
              Object {
                "flexDirection": "row",
                "paddingTop": 2,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 14,
                }
              }
            >
              20 pts
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "paddingTop": 2,
              }
            }
          >
            <DotSeparated
              separated={
                Array [
                  "2 Replies",
                  "1 Unread",
                ]
              }
              style={
                Object {
                  "fontSize": 14,
                }
              }
            />
          </View>
        </View>
        <TouchableHighlight
          accessibilityLabel="Change Dear Ivy Iversen, your last name cannot start with your first name to different grouping"
          accessibilityTraits="button"
          accessible={true}
          activeOpacity={0.85}
          delayPressOut={100}
          onPress={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "flex": 0,
              "flexDirection": "column",
              "height": 43,
              "justifyContent": "center",
              "marginRight": -8,
              "marginTop": -15,
              "width": 43,
            }
          }
          testID="discussion.kabob-1"
          underlayColor="#ffffff00"
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/images/kabob.png",
              }
            }
            style={
              Object {
                "alignSelf": "center",
                "height": 18,
                "tintColor": "#8B969E",
                "transform": Array [
                  Object {
                    "rotate": "180deg",
                  },
                ],
                "width": 18,
              }
            }
          />
        </TouchableHighlight>
      </View>
    </Row>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#008EE2",
        "borderRadius": 2.5,
        "height": 5,
        "left": 8,
        "position": "absolute",
        "top": 6,
        "width": 5,
      }
    }
  />
</View>
`;

exports[`DiscussionsRow renders with no assignment 1`] = `
<View>
  <View
    accessible={false}
  >
    <Row
      accessible={true}
      border="bottom"
      disclosureIndicator={false}
      height="auto"
      onPress={[Function]}
      renderImage={[Function]}
      selected={false}
      testID="DiscussionListCell.1"
      title="Dear Ivy Iversen, your last name cannot start with your first name"
      titleProps={
        Object {
          "ellipsizeMode": "tail",
          "numberOfLines": 2,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={false}
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <DotSeparated
            separated={
              Array [
                "Last post Dec 10, 2016 at 9:03 PM",
              ]
            }
            style={
              Object {
                "color": "#8B969E",
                "fontSize": 14,
                "marginTop": 2,
              }
            }
          />
          <View
            style={
              Object {
                "flexDirection": "row",
                "paddingTop": 2,
              }
            }
          >
            <DotSeparated
              separated={
                Array [
                  "2 Replies",
                  "1 Unread",
                ]
              }
              style={
                Object {
                  "fontSize": 14,
                }
              }
            />
          </View>
        </View>
        <TouchableHighlight
          accessibilityLabel="Change Dear Ivy Iversen, your last name cannot start with your first name to different grouping"
          accessibilityTraits="button"
          accessible={true}
          activeOpacity={0.85}
          delayPressOut={100}
          onPress={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "flex": 0,
              "flexDirection": "column",
              "height": 43,
              "justifyContent": "center",
              "marginRight": -8,
              "marginTop": -15,
              "width": 43,
            }
          }
          testID="discussion.kabob-1"
          underlayColor="#ffffff00"
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/images/kabob.png",
              }
            }
            style={
              Object {
                "alignSelf": "center",
                "height": 18,
                "tintColor": "#8B969E",
                "transform": Array [
                  Object {
                    "rotate": "180deg",
                  },
                ],
                "width": 18,
              }
            }
          />
        </TouchableHighlight>
      </View>
    </Row>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#008EE2",
        "borderRadius": 2.5,
        "height": 5,
        "left": 8,
        "position": "absolute",
        "top": 6,
        "width": 5,
      }
    }
  />
</View>
`;

exports[`DiscussionsRow renders with no unread count 1`] = `
<View>
  <View
    accessible={false}
  >
    <Row
      accessible={true}
      border="bottom"
      disclosureIndicator={false}
      height="auto"
      onPress={[Function]}
      renderImage={[Function]}
      selected={false}
      testID="DiscussionListCell.1"
      title="Dear Ivy Iversen, your last name cannot start with your first name"
      titleProps={
        Object {
          "ellipsizeMode": "tail",
          "numberOfLines": 2,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={false}
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <DotSeparated
            separated={
              Array [
                "Multiple Due Dates",
              ]
            }
            style={
              Object {
                "color": "#8B969E",
                "fontSize": 14,
                "marginTop": 2,
              }
            }
          />
          <View
            style={
              Object {
                "flexDirection": "row",
                "paddingTop": 2,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 14,
                }
              }
            >
              12 pts
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "paddingTop": 2,
              }
            }
          >
            <DotSeparated
              separated={
                Array [
                  "2 Replies",
                  "0 Unread",
                ]
              }
              style={
                Object {
                  "fontSize": 14,
                }
              }
            />
          </View>
        </View>
        <TouchableHighlight
          accessibilityLabel="Change Dear Ivy Iversen, your last name cannot start with your first name to different grouping"
          accessibilityTraits="button"
          accessible={true}
          activeOpacity={0.85}
          delayPressOut={100}
          onPress={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "flex": 0,
              "flexDirection": "column",
              "height": 43,
              "justifyContent": "center",
              "marginRight": -8,
              "marginTop": -15,
              "width": 43,
            }
          }
          testID="discussion.kabob-1"
          underlayColor="#ffffff00"
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/images/kabob.png",
              }
            }
            style={
              Object {
                "alignSelf": "center",
                "height": 18,
                "tintColor": "#8B969E",
                "transform": Array [
                  Object {
                    "rotate": "180deg",
                  },
                ],
                "width": 18,
              }
            }
          />
        </TouchableHighlight>
      </View>
    </Row>
  </View>
  <View />
</View>
`;

exports[`DiscussionsRow renders with points possible 1`] = `
<View>
  <View
    accessible={false}
  >
    <Row
      accessible={true}
      border="bottom"
      disclosureIndicator={false}
      height="auto"
      onPress={[Function]}
      renderImage={[Function]}
      selected={false}
      testID="DiscussionListCell.1"
      title="Dear Ivy Iversen, your last name cannot start with your first name"
      titleProps={
        Object {
          "ellipsizeMode": "tail",
          "numberOfLines": 2,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={false}
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <DotSeparated
            separated={
              Array [
                "Due May 31, 2037 at 11:59 PM",
              ]
            }
            style={
              Object {
                "color": "#8B969E",
                "fontSize": 14,
                "marginTop": 2,
              }
            }
          />
          <View
            style={
              Object {
                "flexDirection": "row",
                "paddingTop": 2,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 14,
                }
              }
            >
              12 pts
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "paddingTop": 2,
              }
            }
          >
            <DotSeparated
              separated={
                Array [
                  "2 Replies",
                  "1 Unread",
                ]
              }
              style={
                Object {
                  "fontSize": 14,
                }
              }
            />
          </View>
        </View>
        <TouchableHighlight
          accessibilityLabel="Change Dear Ivy Iversen, your last name cannot start with your first name to different grouping"
          accessibilityTraits="button"
          accessible={true}
          activeOpacity={0.85}
          delayPressOut={100}
          onPress={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "flex": 0,
              "flexDirection": "column",
              "height": 43,
              "justifyContent": "center",
              "marginRight": -8,
              "marginTop": -15,
              "width": 43,
            }
          }
          testID="discussion.kabob-1"
          underlayColor="#ffffff00"
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/images/kabob.png",
              }
            }
            style={
              Object {
                "alignSelf": "center",
                "height": 18,
                "tintColor": "#8B969E",
                "transform": Array [
                  Object {
                    "rotate": "180deg",
                  },
                ],
                "width": 18,
              }
            }
          />
        </TouchableHighlight>
      </View>
    </Row>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#008EE2",
        "borderRadius": 2.5,
        "height": 5,
        "left": 8,
        "position": "absolute",
        "top": 6,
        "width": 5,
      }
    }
  />
</View>
`;

exports[`DiscussionsRow renders without points possible 1`] = `
<View>
  <View
    accessible={false}
  >
    <Row
      accessible={true}
      border="bottom"
      disclosureIndicator={false}
      height="auto"
      onPress={[Function]}
      renderImage={[Function]}
      selected={false}
      testID="DiscussionListCell.1"
      title="Dear Ivy Iversen, your last name cannot start with your first name"
      titleProps={
        Object {
          "ellipsizeMode": "tail",
          "numberOfLines": 2,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={false}
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <DotSeparated
            separated={
              Array [
                "Due May 31, 2037 at 11:59 PM",
              ]
            }
            style={
              Object {
                "color": "#8B969E",
                "fontSize": 14,
                "marginTop": 2,
              }
            }
          />
          <View
            style={
              Object {
                "flexDirection": "row",
                "paddingTop": 2,
              }
            }
          >
            <DotSeparated
              separated={
                Array [
                  "2 Replies",
                  "1 Unread",
                ]
              }
              style={
                Object {
                  "fontSize": 14,
                }
              }
            />
          </View>
        </View>
        <TouchableHighlight
          accessibilityLabel="Change Dear Ivy Iversen, your last name cannot start with your first name to different grouping"
          accessibilityTraits="button"
          accessible={true}
          activeOpacity={0.85}
          delayPressOut={100}
          onPress={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "flex": 0,
              "flexDirection": "column",
              "height": 43,
              "justifyContent": "center",
              "marginRight": -8,
              "marginTop": -15,
              "width": 43,
            }
          }
          testID="discussion.kabob-1"
          underlayColor="#ffffff00"
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/images/kabob.png",
              }
            }
            style={
              Object {
                "alignSelf": "center",
                "height": 18,
                "tintColor": "#8B969E",
                "transform": Array [
                  Object {
                    "rotate": "180deg",
                  },
                ],
                "width": 18,
              }
            }
          />
        </TouchableHighlight>
      </View>
    </Row>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#008EE2",
        "borderRadius": 2.5,
        "height": 5,
        "left": 8,
        "position": "absolute",
        "top": 6,
        "width": 5,
      }
    }
  />
</View>
`;
